File: problems/338/solution.py
class Solution:
    def countBits(self, num: int) -> List[int]:
        n = [0] * (num + 1)
        
        for i in range(1, num + 1):
            n[i] = n[i - 1] + 1 if i % 2 else n[i // 2]
        
        return n

File: problems/338/solution.go
package main
func countBits(num int) []int {
    n := make([]int, num + 1)
    
    for i := 1; i <= num; i++ {
        if i % 2 == 0 {
            n[i] = n[i/2]
        } else {
            n[i] = n[i - 1] + 1
        }
    }
    
    return n
}